lib/commanded/aggregates/aggregate.ex

```diff
@@ -242,22 +267,39 @@ defmodule Commanded.Aggregates.Aggregate do
           aggregate_lifespan_timeout(lifespan, :after_error, error)
       end
 
+    num_events =
+      case reply do
+        {:ok, events} -> Enum.count(events)
+        _ -> 0
+      end
+
     reply = ExecutionContext.format_reply(reply, context, state)
 
     state = %Aggregate{state | lifespan_timeout: lifespan_timeout}
 
     %Aggregate{aggregate_version: aggregate_version, snapshotting: snapshotting} = state
 
-    if Snapshotting.snapshot_required?(snapshotting, aggregate_version) do
-      :ok = GenServer.cast(self(), :take_snapshot)
+    response =
+      if Snapshotting.snapshot_required?(snapshotting, aggregate_version) do
+        :ok = GenServer.cast(self(), :take_snapshot)
 
-      {:reply, reply, state}
-    else
-      case lifespan_timeout do
-        {:stop, reason} -> {:stop, reason, reply, state}
-        lifespan_timeout -> {:reply, reply, state, lifespan_timeout}
+        {:reply, reply, state}
+      else
+        case lifespan_timeout do
+          {:stop, reason} -> {:stop, reason, reply, state}
+          lifespan_timeout -> {:reply, reply, state, lifespan_timeout}
+        end
       end
-    end
+
+    duration = System.monotonic_time() - start
+
+    :telemetry.execute(
+      [:commanded, :aggregate, :execute, :stop],
```

Comment:
my only reservation here is if there should be a separate event for errors

Reply:
It would be useful to capture exceptions/errors, but that can be added in a separate PR.

Reply:
noted.